### ISSUE TITLE
Unable to build core 

### ISSUE THREAD

**Issue by yogeshjain19061988** [COVESA/capicxx-core-tools#35]
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.genivi.commonapi.core.feature.feature.group 3.2.0.qualifier
[ERROR]   Missing requirement: org.genivi.commonapi.core.ui 3.2.0.qualifier requires 'bundle com.google.inject [3.0.0,4.0.0)' but it could not be found
[ERROR]   Cannot satisfy dependency: org.genivi.commonapi.core.feature.feature.group 3.2.0.qualifier depends on: org.genivi.commonapi.core.ui 0.0.0

**Comment by yangguangdelta** [comment]
yes, same error, I tried to delete bundle-version [3.0.0,4.0.0) or change it to [3.0.0,5.1.0) in MANIFEST.MF, build successfully, but the tool still not work when convert fidl to source files.  I think we need contributor's help to resolve it.

**Comment by zhhhhhhhh** [comment]
me too ,15 days pass,have any good?

[INFO] Resolving dependencies of MavenProject: commonapi:org.genivi.commonapi.core.feature:3.2.0-SNAPSHOT @ /home/ubuntu/capicxx-core-tools/org.genivi.commonapi.core.feature/pom.xml
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.genivi.commonapi.core.feature.feature.group 3.2.0.qualifier
[ERROR]   Missing requirement: org.genivi.commonapi.core.ui 3.2.0.qualifier requires 'bundle com.google.inject [3.0.0,4.0.0)' but it could not be found
[ERROR]   Cannot satisfy dependency: org.genivi.commonapi.core.feature.feature.group 3.2.0.qualifier depends on: org.genivi.commonapi.core.ui 0.0.0
[ERROR]
[ERROR] See http://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.
[ERROR] Cannot resolve dependencies of MavenProject: commonapi:org.genivi.commonapi.core.feature:3.2.0-SNAPSHOT @ /home/ubuntu/capicxx-core-tools/org.genivi.commonapi.core.feature/pom.xml: See log for details -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionExcept

**Comment by livingsilver94** [comment]
As a workaround, I added this snippet to `org.genivi.commonapi.core.ui/pom.xml`:

```xml
<dependencies>
  <dependency>
      <groupId>com.google.inject</groupId>
      <artifactId>guice</artifactId>
      <version>3.0</version>
  </dependency>
</dependencies>
```

And removed the offending line from `MANIFEST.MF`.

**Comment by yogeshjain19061988** [comment]
> As a workaround, I added this snippet to `org.genivi.commonapi.core.ui/pom.xml`:
> 
> ```
> <dependencies>
>   <dependency>
>       <groupId>com.google.inject</groupId>
>       <artifactId>guice</artifactId>
>       <version>3.0</version>
>   </dependency>
> </dependencies>
> ```
> 
> And removed the offending line from `MANIFEST.MF`.

what lines have you removed from the MANIFEST.MF file? can you put the MANIFEST file here ?

**Comment by livingsilver94** [comment]
@yogeshjain19061988 why did you close this issue? I posted a workaround; it's not a solution. I may provide a PR if my workaround is decent enough to be an upstream patch, but I'm not a Java developer so I'll leave this consideration to you.

Anyway, `org.genivi.commonapi.core.ui/META-INF/MANIFEST.MF`. Just remove this line:

`com.google.inject;bundle-version="[3.0.0,4.0.0)";visibility:=reexport,`

**Comment by yangguangdelta** [comment]
@livingsilver94 building is successful for your workaround, but the tool still not work when convert fidl to source files

**Comment by zhhhhhhhh** [comment]
might ,you guys can download that has been built generator from hub. but i only find 3.12,not newest

**Comment by yogeshjain19061988** [comment]
> @yogeshjain19061988 why did you close this issue? I posted a workaround; it's not a solution. I may provide a PR if my workaround is decent enough to be an upstream patch, but I'm not a Java developer so I'll leave this consideration to you.
> 
> Anyway, `org.genivi.commonapi.core.ui/META-INF/MANIFEST.MF`. Just remove this line:
> 
> `com.google.inject;bundle-version="[3.0.0,4.0.0)";visibility:=reexport,`

by mistake, it got closed.

**Comment by xiaoqishuier** [comment]
Many solutions have been tried. Is there any new progress

**Comment by martinbrook** [comment]
Also interested

**Comment by Stefan3934** [comment]
I face the same build issue: "[ERROR] Missing requirement: org.genivi.commonapi.core.ui 3.2.0.qualifier requires 'bundle com.google.inject [3.0.0,4.0.0)' but it could not be found"

I tried on on Ubuntu 18.04.1 and 22.04.1 (openjdk version "1.8.0_342")

The change in org.genivi.commonapi.core.ui/META-INF/MANIFEST.MF, mentioned above, helps to get it built, but then the tool has run-time errors. [1663320338987.log](https://github.com/COVESA/capicxx-core-tools/files/9597612/1663320338987.log)

Any help appreciated.

**Comment by Gustavo1515** [comment]
Issue fixed with PR #50 

Cc: @goncaloalmeida
